### DISCUSSION TITLE
policy.xml location for manual build and install

### DISCUSSION THREAD

**Discussion by bfcoder** [ImageMagick/ImageMagick#3753] [Help] [answered]
I have manually built ImageMagick 7 with HEIC support enabled. This is on an Ubuntu 20.04 VM.

My version:
```
# convert -version
Version: ImageMagick 7.0.11-13 Q16 x86_64 2021-05-17 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): bzlib djvu fontconfig freetype heic jbig jng jpeg lcms lqr lzma openexr pangocairo png tiff webp x xml zlib
```

My problem is it doesn't have a policy to use.
```
# convert -list policy

Path: [built-in]
  Policy: Undefined
    rights: None
```

As I understand, it should have a policy listed in there. Typically in /etc/ImageMagic-7/policy.xml or /usr/local/etc/ImageMagick-7/policy.xml if I understand correctly. I actually do see the policy.xml located in the /usr/local/etc/ImageMagick-7/ directory. But alas it isn't being utilized by ImageMagick.

I utilized parts of https://eplt.medium.com/5-minutes-to-install-imagemagick-with-heic-support-on-ubuntu-18-04-digitalocean-fe2d09dcef1 and also parts of https://www.everclimb.co/blog/how-to-install-imagemagick-7-on-ubuntu-16-04 to do the build.

Basically here are the steps I took:
* added the ppa's for libheif and libde265
* enabled the universe deb-src sources
* installed autoconf, libtool, libde265-dev, libheif-dev
* build-dep for imagemagick, libmagickcore-dev
* download https://www.imagemagick.org/download/ImageMagick.tar.gz and unpack it
* configure with `./configure --with-heic=yes`
* then `make && make install`
* and finally `ldconfig`

I don't know how to tell it to use a policy.xml now. And I haven't been successful in finding any documentation that states how to do it. Can anyone point me in the right direction?

Thanks!

**Comment by snibgo** [comment, marked as answer]
> I actually do see the policy.xml located in the /usr/local/etc/ImageMagick-7/ directory. But alas it isn't being utilized by ImageMagick.

How do you know it isn't being used? "-debug all" will tell you.

What is in your policy.xml? Perhaps it contains only comments.

policy.xml is used to restrict rights. It doesn't _enable_ any rights, it only _removes_ them. What rights do you want to remove?

**Reply by bfcoder** [reply]
When I do a `convert -list policy` it should list the policy being used. Correct? It doesn't list anything.

```
# convert -list policy -debug all
2021-05-28T16:57:34+00:00 0:00.000 0.000u 7.0.11 Configure convert[37189]: utility.c/ExpandFilenames/969/Configure
  Command line: convert {-list} {policy} {-debug} {all}

Path: [built-in]
  Policy: Undefined
    rights: None 
```

**Reply by bfcoder** [reply]
Oh, I see. Since everything was commented out in `/usr/local/etc/ImageMagick-7/policy.xml` it just doesn't show anything. I uncommented a line and now see this:

```
# convert -debug all -list policy
2021-05-28T17:39:18+00:00 0:00.000 0.000u 7.0.11 Configure convert[38280]: utility.c/ExpandFilenames/969/Configure
  Command line: convert {-debug} {all} {-list} {policy}

Path: /usr/local/etc/ImageMagick-7/policy.xml
  Policy: Module
    rights: None 
    pattern: {PS,PDF,XPS}

Path: [built-in]
  Policy: Undefined
    rights: None 
```

I was expecting it to show the policy.xml file being used no matter what. Thanks!

**Comment by snibgo** [comment]
My mistake, I thought that "-debug all" would show the path to the policy.xml, but it doesn't. As you have discovered, "-list policy" shows the path to any policy.xml files that actually restrict policy.

(In general, there might be two policy.xml files in use: a general one for all users, and one for each particular user.)

You have IM v7, so I suggest you use "magick", not "convert".

**Reply by bfcoder** [reply]
Awesome. Thank you again for your help!
